pkg/apis/serving/v1alpha1/metadata_validation.go

```diff
@@ -30,11 +30,22 @@ import (
 // resources is correct.
 func ValidateObjectMetadata(meta metav1.Object) *apis.FieldError {
 	name := meta.GetName()
+	generateName := meta.GetGenerateName()
 
-	msgs := validation.IsDNS1035Label(name)
-	if len(msgs) > 0 {
+	if name != "" {
+		msgs := validation.IsDNS1035Label(name)
```

Comment:
Does this work? Since generate name most likely is going to have a trailing "-" and DNS1035 label wants a terminating letter, iirc.
In addition, if this name matches the label, the generated name might not (it'll be longer).

Reply:
I'm not validating `generateName` for that reason

Reply:
Some API docs say `generatedName` gets truncated to fit the suffix

Reply:
🤦‍♂️ 
I misread.
But should we still do some validation thereof? I.e. truncate the  "-"  and then match?

Reply:
> But should we still do some validation thereof?

I don't want to go too far down the rabbit hole of figuring out what the API server does exactly. Validating `generatedName` could be a separate PR.

Trying to unblock https://github.com/knative/serving/pull/3179 at this point.

Reply:
Related issue: https://github.com/knative/serving/issues/3182